azurerm/data_source_builtin_role_definition.go

```diff
@@ -0,0 +1,102 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func dataSourceArmBuiltInRoleDefinition() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmBuiltInRoleDefinitionRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Contributor",
+					"Reader",
+					"Owner",
+					"VirtualMachineContributor",
+				}, false),
+			},
+
+			// Computed
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"permissions": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"actions": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"not_actions": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"assignable_scopes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceArmBuiltInRoleDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).roleDefinitionsClient
+	name := d.Get("name").(string)
+	roleDefinitionIds := map[string]string{
+		"Contributor":               "/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c",
```

Comment:
Are these values (`Contributor`, `Owner`, Reader`, `VirtualMachineContributor`) specific to our account or azure in general?

Reply:
They're Azure in General - this Data Source is mostly there so users don't have to refer to GUIDs tbh

Reply:
Perfect!